lib/ansible/modules/windows/win_find.ps1

```diff
@@ -0,0 +1,347 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+
+$paths = Get-AnsibleParam -obj $params -name 'paths' -failifempty $true
+
+$age = Get-AnsibleParam -obj $params -name 'age' -failifempty $false -default $null
+$age_stamp = Get-AnsibleParam -obj $params -name 'age_stamp' -failifempty $false -default 'mtime' -ValidateSet 'mtime','ctime','atime'
+$file_type = Get-AnsibleParam -obj $params -name 'file_type' -failifempty $false -default 'file' -ValidateSet 'file','directory'
+$follow = Get-AnsibleParam -obj $params -name 'follow' -type "bool" -failifempty $false -default $false
+$hidden = Get-AnsibleParam -obj $params -name 'hidden' -type "bool" -failifempty $false -default $false
+$patterns = Get-AnsibleParam -obj $params -name 'patterns' -failifempty $false -default $null
+$recurse = Get-AnsibleParam -obj $params -name 'recurse' -type "bool" -failifempty $false -default $false
+$size = Get-AnsibleParam -obj $params -name 'size' -failifempty $false -default $null
+$use_regex = Get-AnsibleParam -obj $params -name 'use_regex' -type "bool" -failifempty $false -default $false
+$get_checksum = Get-AnsibleParam -obj $params -name 'get_checksum' -type "bool" -failifempty $false -default $true
+$checksum_algorithm = Get-AnsibleParam -obj $params -name 'checksum_algorithm' -failifempty $false -default 'sha1' -ValidateSet 'md5', 'sha1', 'sha256', 'sha384', 'sha512'
```

Comment:
So something I would do (but it is not mandatory) is to add -type to everything.
- age_stamp could be -type "int"
- file_type could be -type "str"

We don't handle a list of paths yet, but it could be useful to run Expand_Environment on every path you consider to use, so people can take advantage of %userprofile% or %public%.

I would drop `-failifempty $false` because that's the default.

I would drop `-default $null` because that's the default.

I personally feel that only adding the non-default options is easier to see which options are e.g. required or have a default.

All of this we can do after this is merged, as I don't want delay merging any longer.